app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -0,0 +1,108 @@
+package com.example.bottomnav;
+
+import org.eclipse.cdt.core.dom.ast.IASTDeclaration;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.dom.ast.gnu.cpp.GPPLanguage;
+import org.eclipse.cdt.core.index.IIndex;
+import org.eclipse.cdt.core.parser.*;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parse {
+    public HashMap<String, Contents> constRepo = new HashMap<>();
+    public HashMap<String, ArrayList> structRepo = new HashMap<>();
+
+    public Parse(String code) throws Exception {
+        IASTTranslationUnit translationUnit = getIASTTranslationUnit(code.toCharArray());
+        process(translationUnit);
+    }
+
+    private void process(IASTNode node) throws Exception {
+        IASTNode[] children = node.getChildren();
+        for (IASTNode child : children) {
+            if (child instanceof CPPASTSimpleDeclaration) {
+                CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) child;
+
+                if (declaration.getDeclSpecifier() instanceof CPPASTCompositeTypeSpecifier) {
+                    CPPASTCompositeTypeSpecifier compSpec = (CPPASTCompositeTypeSpecifier) declaration.getDeclSpecifier();
+                    CPPASTName name = (CPPASTName) compSpec.getName();
+                    saveStuct(compSpec.getDeclarations(false), name.getRawSignature());
+                } else {
+                    saveConst(child);
+                }
+
+            }
+        }
+    }
+
+    private ArrayList getComponents(IASTNode node) throws Exception {
```

Comment:
May be helpful to have documentation for what this does, and the assumptions of the inputs (here, the input needs to be a CPPASTSimpleDeclaration). Really, what this seems to be doing is trying to parse a CPPASTSimpleDeclaration into a name, type, type-qualifier, and initializer.

Also highly recommend having another struct-like type that saves the type of the stuff you're packing into the output, instead of an ArrayList takes any Object (and only knows it as Objects). Having items indexed by a name is also cleaner than having them indexed by a number, especially since each slot in the ArrayList has a different meaning which is count to typical usage.